third_party/terraform/data_sources/data_source_google_storage_bucket_object.go

```diff
@@ -24,6 +26,12 @@ func dataSourceGoogleStorageBucketObjectRead(d *schema.ResourceData, meta interf
 	bucket := d.Get("bucket").(string)
 	name := d.Get("name").(string)
 
+	// URL encode folder names, but to ensure backward compatibility don't url encode
+	// them if they were already encoded manually in config.
+	// see https://github.com/terraform-providers/terraform-provider-google/issues/3176
```

Comment:
lgtm though I think we should probably encode it regardless - i.e. if they didn't encode it correctly and just did part of it. I guess though they probably have it right if they are currently using it, so either way.

Reply:
Yup, if they had `name = "folder/folder%2Fstuff.jpg" it wouldn't work now or after this change.